include/highfive/H5PropertyList.hpp

```diff
@@ -133,6 +133,27 @@ class PropertyListBase: public Object {
     friend T details::get_plist(const U&, hid_t (*f)(hid_t));
 };
 
+/// \interface PropertyInterface
+/// \brief HDF5 file property object
+///
+/// A property is an object which is expected to have a method with the
+/// following signature `void apply(hid_t hid) const`
+///
+/// \sa Instructions to document C++20 concepts with Doxygen: https://github.com/doxygen/doxygen/issues/2732#issuecomment-509629967
+///
+/// \cond
+#if __cplusplus >= 202002L
```

Comment:
@antonysigma  I would like there to be a way of forcing the concepts to be off. I'm worried a user will find a compiler that claims to be `202002L` but doesn't support concepts; or has a compiler bug. If this happens they would be unable to use HighFive, because a nice, but non-essential feature is broken.

If you like please add a CMake variable `HIGHFIVE_HAS_CONCEPTS` which defaults to `On`. Then add code to instruct the compiler to add `-DHIGHFIVE_HAS_CONCEPTS={0,1}` has appropriate. Then, the if-condition above would be
```
#if HIGHFIVE_HAS_CONCEPTS && __cplusplus >= 202002L
```

Reply:
Done. I am not familiar to CMake >12.0 though. Feel free to revise the changes where it make sense.

Reply:
Thanks! I slightly changed it and tested if it works. One might need `-fconcepts-diagnostics-depth=2` to get meaningful compiler output.

Reply:
Great! Let me know if you need anything else. I will standby.

Reply:
I'd like to prepare a PR which modifies CI to also compile HighFive as C++20 code. Otherwise, the concepts aren't tested during CI. Then we can rebase this PR onto the other one; and everything is tested nicely again. 

Thank you for this nice PR.